### PR TITLE
[IMP] hr_timeoff,web: convert radio image field

### PR DESCRIPTION
**Pull request by mcm-odoo**
This commit converts the radio image field of hr_timeoff
to the new framework

### CODE REVIEW

File: addons/hr_holidays/static/src/radio_image_field/radio_image_field.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+/** @odoo-module **/
+
+import { registry } from "@web/core/registry";
+import { RadioField } from "@web/views/fields/radio/radio_field";
+
+class RadioImageField extends RadioField {}
+RadioImageField.template = "hr_holidays.RadioImageField";
+
+registry.category("fields").add("radio_image", RadioImageField);
```
Comment (aab-odoo) [inline addons/hr_holidays/static/src/radio_image_field/radio_image_field.js:9]: Don't change this in this PR, but I think that addons defining views/widgets should prefix the key in the registry with the addon name

File: addons/hr_holidays/static/src/radio_image_field/radio_image_field.xml [hunk 2]
```diff
@@ -0,0 +1,36 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<templates id="template" xml:space="preserve">
+
+    <t t-name="hr_holidays.RadioImageField" owl="1">
+        <div role="radiogroup" class="d-flex flex-wrap" t-att-aria-label="string">
```
Comment (aab-odoo) [inline addons/hr_holidays/static/src/radio_image_field/radio_image_field.xml:5]: radiogroup even in readonly?

File: addons/hr_holidays/static/src/radio_image_field/radio_image_field.js (hunk 1 above)
Comment (aab-odoo) [reply addons/hr_holidays/static/src/radio_image_field/radio_image_field.js:9]: Maybe worth notifying them (and others doing the same)

File: addons/web/static/src/views/fields/radio/radio_field.js [hunk 3]
```diff
@@ -57,7 +57,7 @@ RadioField.defaultProps = {
 RadioField.displayName = _lt("Radio");
 RadioField.supportedTypes = ["many2one", "selection"];
 
-RadioField.isEmpty = () => false;
+RadioField.isEmpty = (record, fieldName) => record.data[fieldName] === false;
```
Comment (aab-odoo) [inline addons/web/static/src/views/fields/radio/radio_field.js:60]: This looks like a fix, if you know what I mean :D

**Comment by aab-odoo** [comment]
Ho also, is it fine to remove legacy code? The view using it never has a js_class (even with an extension from another addon)? And is it displayed correctly in studio?

**Comment by mcm-odoo** [comment]
@aab-odoo I've kept the legacy code, the basic_relational_model needs the legacy field to know that we want special data.
I don't know if I can add some tests in this module.

**Review by aab-odoo — APPROVED** [review]
robodoo r+

**Pull request closed without merging**